### PR TITLE
.NET + Azure Cognitive Services blog post

### PR DESCRIPTION
**Pull request by alisaduncan**
This blog post has the following features:

- [x] A GitHub Repository with a polished README
- [x] A GitHub Repository under the github.com/oktadev account
- [x] A title that's approved by Dev Advocacy
- [x] A URL that's approved by Dev Advocacy
- [x] The content has been run through Grammarly (https://www.grammarly.com/)
- [x] Rendered locally and confirmed that no Markdown typos exist

### CODE REVIEW

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
In this article, you will learn how to use the [Face service](https://docs.microsoft.com/en-us/azure/cognitive-services/face/) library to perform facial analysis in a .NET MVC application and store user profile pictures in [Azure Blob Container Storage](https://azure.microsoft.com/en-us/services/storage/blobs/). You'll also authenticate with Okta and store user data as custom profile attributes.
```
Not seeing any reference to "Vision" per se.

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
At the end of this post, you'll be able to upload a profile picture in your app and get information about image error conditions, such as when zero or more than one face is detected or when your facial features don't match a new picture.
```
clarity

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
You'll use Okta to secure the application quickly with Okta SDKs, so you don't have to spin up an identity provider and deal with the tricky details of authentication.
```
clarity, conciseness

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (alisaduncan) [reply _source/_posts/2021-12-20-net-azure-cognitive-services.md]: @havi I was drawing from Azure Cognitive Services site, where they define service categories (Speech, Language, Vision, Decision) to match with the last sentence of the paragraph above.

But maybe it doesn't have to be that specific and "Face AP"I is enough? (I don't know how I feel about "Face library"). Here's a screenshot of Azure's site with their offerings categorized. Thoughts?

image

Comment (Quorralyne) [reply _source/_posts/2021-12-20-net-azure-cognitive-services.md]: "In this article, you will learn how to use the Vision [Face API](https://docs.microsoft.com/en-us/azure/cognitive-services/face/)"

Is my two cents.

Comment (havi) [reply _source/_posts/2021-12-20-net-azure-cognitive-services.md]: Thanks both! I think Heather's two cents nails it  - I didn't spot the word "Vision" anywhere when I landed on the "cognitive-services/face/" page so I thought that naming might have changed.

**Review by Quorralyne — APPROVED** [review]
Approved after merging Havi's edits - with my one alternate suggestion :)

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
To make your profile picture viewable, we need to persist the image somewhere. We'll use Azure Storage to do this and use a custom attribute in Okta to store a unique identifier associated with the file in Azure. You'll be able to upload and display a profile picture in the app.
```
clarity

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
Log in to the [Okta admin dashboard](https://developer.okta.com/login/). Then navigate to **Directory** > **Profile Editor**. You should see an item named **User (default)** in the list, which is Okta's default user profile template. Select **User (default)** to open the **Profile Editor**. Press the **+Add Attribute** button.
```
add space

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
Once you have an Azure subscription, open the [Azure Portal](https://portal.azure.com). Open the menu by pressing the *hamburger menu* on the left, and select **Storage accounts**. Press **+Create** to create a storage account. In the **Create a storage account** view, create a Resource Group if you don't already have one for test projects. I named my Resource Group "OktaDemo". Enter `oktaprofilepicture` as the storage name, select a region (Azure's default selection is fine here), then select **Standard** performance and **Locally-redundant storage (LRS)** for redundancy. Press the **Review + create** button at the bottom as the default options for the remaining selections are acceptable. Press **Create** to complete creating the storage account.
```
italicize for anyone left on earth who doesn't know this has nothing to do with beef
remove an "and"

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
Next, we'll need to add the packages and set up the service. Add two Nuget packages — one for Azure Storage and the other for integrating Azure clients into the DI system. Add the packages by running the following commands in the terminal.
```
avoids the repetition of "You'll need"

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
Update the `Profile()` and `EditProfile(UserProfileViewModel profile)` methods to upload and view the blob container image. First, let's update the `Profile()` method. 
```
conciseness

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
Sometimes you need to clear everything out and start over. I uploaded a picture of myself wearing oversized sunglasses and a hat, and the face service no longer recognized me. 😎
```
avoids the passive voice here for better readability

File: _source/_posts/2021-12-20-net-azure-cognitive-services.md
Comment (havi) [inline _source/_posts/2021-12-20-net-azure-cognitive-services.md]:
```suggestion
I hope the tutorial was interesting and enjoyable for you. You can get the [full source code of the project from GitHub](https://github.com/oktadev/okta-dotnet-azure-cognitive-services-example). For more ASP.NET Core and Okta articles, check out these posts:
```
conciseness, avoids the dangling "it" at the end.

**Review by havi — CHANGES_REQUESTED** [review]
minor tweaks for readability. thanks.

**Review by havi — APPROVED** [review]
LGTM!

**Pull request merged**
